src/dune/__main__.py

```diff
@@ -93,6 +93,18 @@ def load_all_modules_from_dir(plugin_dir):
             WAS_REMAINDER_ARGS_USED = False
 
             if args.start is not None:
+                WAS_REPLAY_BLOCKCHAIN_USED = False
+                if args.rmdirtydb is True:
+                    node_state_dir = '/app/nodes/' + args.start[0] +'/state/'
+                    if dune_sys.docker.dir_exists(node_state_dir) and not dune_sys.is_named_node_running(args.start[0]):
+                        print("Use --replay-blockchain option of nodeos to clean "
+                                + "a database dirty flag for node [" + args.start[0] + "]...\n")
+                        WAS_REPLAY_BLOCKCHAIN_USED = True
+                    elif dune_sys.is_named_node_running(args.start[0]):
+                        print("no action for --rmdirtydb, because node [ "  + args.start[0]  + " ] is already running\n")
+                    else:
+                        print("no action for --rmdirtydb, because container directory "  + node_state_dir + " does not exist\n")
```

Comment:
Maybe, I'm looking at this wrong, but doesn't this always replay-blockchain on start even if there is no dirty db? I thought the idea was to add an option to auto replay if there was a dirty db, not always replay.

Reply:
I think you are? These are warnings for the user and are in this branch:
```python
 if args.rmdirtydb is True:
```

`WAS_REPLAY_BLOCKCHAIN_USED` is set false at the entry and ONLY if the node is NOT running.

But the logic could be cleaned up a little bit I think.

Reply:
See my new code for detecting dirty db flag in nodeos